### PR TITLE
Retrospective

### LINKED ISSUE #9: Writing the retrospective document

### PR DESCRIPTION
**Pull request by Safaishag**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [ ] The file name describes the function's behavior
- [ ] There is a module docstring in the function file
- [ ] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [ ] There is a module docstring in the tests file

### Unit Tests

- [ ] The test class has a helpful name in PascalCase
- [ ] The test class has a docstring
- Every unit test has
  - [ ] A helpful name
  - [ ] A clear docstring
  - [ ] Only one assertion
  - [ ] There is no logic in the unit test
- [ ] All tests pass
- [ ] There are tests for defensive assertions
- [ ] There are tests for boundary cases

### Function Docstring

- [ ] The function's behavior is described
- The function's arguments are described:
  - [ ] Type
  - [ ] Purpose
  - [ ] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [ ] Type
  - [ ] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [ ] Each assumption about an argument is checked with an assertion
  - [ ] Each assertion checks for _only one_ assumption about the argument
- [ ] Include 3 or more (passing!) doctests

### The Function

- [ ] The function's name describes it's behavior
- [ ] The function's name matches the file name
- [ ] The function has correct type annotations

### Do's

- [ ] Variable names help to understand the strategy
- [ ] Any comments are clear and describe the strategy
- [ ] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [ ] The function's strategy _is not_ described in the documentation
- [ ] Comments explain the _strategy_, **not** the _implementation_
- [ ] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [ ] The code passes the formatting checks
- [ ] The code passes all Ruff linting checks
- [ ] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [ ] Variables are named with snake_case
- [ ] Variable names are clear and helpful
- [ ] The code follows the strategy as simply as possible
- [ ] The implementation is as simple as possible given the strategy
- [ ] The code includes defensive assertions
- [ ] Defensive assertions include as little logic as possible

### CODE REVIEW

File: collaboration/retrospective.md [hunk 1]
```diff
@@ -1,23 +1,130 @@
...
+- Improve overall efficiency in managing workflows and deadlines.
+
+---
+
+### **Final Note**
+
+The success of this project was a testament to the team's dedication,
+openness, and collaborative spirit. By building on these strengths and
+ addressing the identified areas of improvement, future projects can be
+ even more effective and rewarding.
```
Comment (Safi222) [inline collaboration/retrospective.md:130]: The document is well-organized!
great job!

File: collaboration/retrospective.md [hunk 2]
```diff
@@ -1,23 +1,130 @@
...
+  to support one another.
+- **Friendliness**: Getting to know the team was easy due to their
+  approachable and collaborative nature.
 
-### Or remove extra steps?
+---
+
+## **Team Ratings**
+
+- **Communication and Collaboration**: 4.55/5 average rating.
```
Comment (Safi222) [inline collaboration/retrospective.md:55]: Aggregating the Google form responses is done in a great way!  reflecting the average opinion of all the team members.

File: collaboration/retrospective.md [hunk 3]
```diff
@@ -1,23 +1,130 @@
...
+- Enhance coding and problem-solving skills.
+- Refine time management strategies.
+- Explore more challenging exercises to deepen understanding of key concepts.
+- Encourage frequent check-ins and cross-team collaboration to
+   address potential roadblocks early.
+
+### **New Skills to Learn**
+
+- Gain deeper proficiency in advanced coding techniques.
+- Improve overall efficiency in managing workflows and deadlines.
```
Comment (Safi222) [inline collaboration/retrospective.md:121]: All points are addressed in comprehensive way.

**Review by Safi222 — APPROVED** [review]
Great work!
The document is well-organized and **effectively outlines the retrospectives of all team members in relation to their responses.**

File: collaboration/retrospective.md (hunk 2 above)
Comment (Omnia-Agabani) [reply collaboration/retrospective.md:55]: It is really showing all the team points and openion

File: collaboration/retrospective.md (hunk 1 above)
Comment (Omnia-Agabani) [inline collaboration/retrospective.md:130]: all points has been added well and professional

**Review by Omnia-Agabani — COMMENTED** [review]
Well Done

**Review by Omnia-Agabani — COMMENTED** [review]
It was good idea to create a google form to take all team member opinion
